doc/internal/man3/ossl_method_construct.pod

```diff
@@ -0,0 +1,134 @@
+=pod
+
+=head1 NAME
+
+OSSL_METHOD_CONSTRUCT_METHOD, ossl_method_construct
+- generic method constructor
+
+=head1 SYNOPSIS
+
+ #include "internal/core.h"
+
+ struct ossl_method_construct_method_st {
+     /* Create store */
+     void *(*alloc_tmp_store)(void);
+     /* Remove a store */
+     void (*dealloc_tmp_store)(void *store);
+     /* Get an already existing method from a store */
+     void *(*get)(OPENSSL_CTX *libctx, void *store, const char *propquery,
+                  void *data);
+     /* Store a method in a store */
+     int (*put)(OPENSSL_CTX *libctx, void *store, const char *propdef,
+                void *method, void *data);
+     /* Construct a new method */
+     void *(*construct)(const OSSL_DISPATCH *fns, OSSL_PROVIDER *prov,
+                        void *data);
+     /* Destruct a method */
+     void (*destruct)(void *method);
+ };
+ typedef struct ossl_method_construct_method OSSL_METHOD_CONSTRUCT_METHOD;
+
+ void *ossl_method_construct(OPENSSL_CTX *ctx, int operation_id,
+                             const char *name, const char *properties,
+                             int force_cache,
+                             OSSL_METHOD_CONSTRUCT_METHOD *mcm, void *mcm_data);
+
+=head1 DESCRIPTION
+
+All libcrypto sub-systems that wants to create their own methods based
+on provider dispatch tables need to do so in exactly the same way.
+ossl_method_construct() does this while leaving it to the sub-systems
+to define more precisely how the methods are created, stored, etc.
+
+=head2 Functions
+
+ossl_method_construct() creates a method by asking all available
+providers for a dispatch table given an C<operation_id>, an algorithm
+C<name> and a set of C<properties>, and then calling appropriate
+functions given by the sub-system specific method creator through
+C<mcm> and the data in C<mcm_data> (which is just passed by
+ossl_method_construct()).
+
+=head2 Structures
+
+A central part in constructing a sub-system specific method is to give
+ossl_method_construct a set of functions, all in the
+C<OSSL_METHOD_CONSTRUCT_METHOD> structure, which holds the following
+function pointers:
+
+=over 4
+
+=item alloc_tmp_store()
+
+Create a temporary method store.
+This store is used to temporarly store methods for easier lookup, for
```

Comment:
Temporarily, there is a missing ‘i’

Reply:
(I'll never be able to remember...)

Fixed